policyengine_us/variables/gov/states/ct/tax/income/subtractions/pension_annuity/ct_pension_annuity_subtraction.py

```diff
@@ -7,17 +7,37 @@ class ct_pension_annuity_subtraction(Variable):
     label = "Connecticut pension and annuity subtraction"
     unit = USD
     definition_period = YEAR
-    defined_for = "ct_pension_annuity_subtraction_eligible"
+    defined_for = StateCode.CT
 
     def formula(tax_unit, period, parameters):
         person = tax_unit.members
-        head = person("is_tax_unit_head", period)
-        spouse = person("is_tax_unit_spouse", period)
-        rate = parameters(
+        filing_status = tax_unit("filing_status", period)
+        status = filing_status.possible_values
+        head_or_spouse = person("is_tax_unit_head_or_spouse", period)
+        agi = tax_unit("adjusted_gross_income", period)
+
+        p = parameters(
             period
-        ).gov.states.ct.tax.income.subtractions.pensions_or_annuity.rate
-        head_or_spouse = head | spouse
+        ).gov.states.ct.tax.income.subtractions.pensions_or_annuity
+
+        # Get the rate based on AGI and filing status
+        is_joint = (filing_status == status.JOINT) | (
```

Comment:
technically it's only joint, but CT allows surviving spouse filers to filer as joint, so this is ok but let's come back to it following https://github.com/PolicyEngine/policyengine-us/issues/6006